src/compiler/checker.ts

```diff
@@ -5976,6 +5977,13 @@ namespace ts {
                 }
 
                 if (targetType) {
+                    if (!assumeTrue) {
+                        if (type.flags & TypeFlags.Union) {
+                            return getUnionType(filter((<UnionType>type).types, t => !isTypeSubtypeOf(t, targetType)));

```

Comment:
@RyanCavanaugh what happened with the idea that if the RHS type isn't a subtype of any types in the union, that we just discard the LHS type and take the RHS type? Is that something that we eventually went with?